### PR TITLE
chore: migrate AuthenticationWebView to hooks

### PR DESCRIPTION
**Pull request by reinaldonetof**
## Proposed changes

migrate AuthenticationWebView to hooks

## How to test or reproduce

- Try to login using Iframe, Saml, SSL

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Documentation update (if none of the other choices apply)

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [X] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [X] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

[NATIVE-7]

[NATIVE-7]: https://rocketchat.atlassian.net/browse/NATIVE-7?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/views/AuthenticationWebView.tsx [hunk 1]
```diff
@@ -43,95 +43,54 @@ window.addEventListener('popstate', function() {
...
 			const data = await fetch(Accounts_Iframe_api_url, { method: Accounts_Iframe_api_method }).then(response => response.json());
 			const resume = data?.login || data?.loginToken;
 			if (resume) {
-				this.login({ resume });
+				login({ resume });
 			}
 		},
 		3000,
-		true
+		{ leading: true }
```
Comment (reinaldonetof) [inline app/views/AuthenticationWebView.tsx:88]: https://css-tricks.com/debouncing-throttling-explained-examples/#aa-leading-edge-or-immediate

File: app/views/AuthenticationWebView.tsx [hunk 2]
```diff
@@ -43,95 +43,54 @@ window.addEventListener('popstate', function() {
...
+const AuthenticationWebView = () => {
+	const [logging, setLogging] = useState(false);
+	const [loading, setLoading] = useState(false);
 
-interface IState {
-	logging: boolean;
-	loading: boolean;
-}
+	const navigation = useNavigation<StackNavigationProp<OutsideModalParamList, 'AuthenticationWebView'>>();
+	const { authType, url, ssoToken } = useRoute<RouteProp<OutsideModalParamList, 'AuthenticationWebView'>>().params;
```
Comment (GleidsonDaniel) [inline app/views/AuthenticationWebView.tsx:51]:
```suggestion
	const { params: {authType, url, ssoToken} } = useRoute<RouteProp<OutsideModalParamList, 'AuthenticationWebView'>>();
```

File: app/views/AuthenticationWebView.tsx [hunk 3]
```diff
@@ -173,39 +132,35 @@ class AuthenticationWebView extends React.PureComponent<IAuthenticationWebView,
...
+		<>
+			<StatusBar />
+			<WebView
+				source={{ uri: url }}
+				userAgent={userAgent}
+				// https://github.com/react-native-community/react-native-webview/issues/24#issuecomment-540130141
+				onMessage={({ nativeEvent }) => onNavigationStateChange(nativeEvent)}
+				onNavigationStateChange={onNavigationStateChange}
+				injectedJavaScript={isIframe ? injectedJavaScript : undefined}
+				onLoadStart={() => {
```
Comment (GleidsonDaniel) [inline app/views/AuthenticationWebView.tsx:154]:
```suggestion
				onLoadStart={() => setLoading(true)}
```

File: app/views/AuthenticationWebView.tsx [hunk 4]
```diff
@@ -173,39 +132,35 @@ class AuthenticationWebView extends React.PureComponent<IAuthenticationWebView,
...
+					setLoading(true);
+				}}
+				onLoadEnd={() => {
```
Comment (GleidsonDaniel) [inline app/views/AuthenticationWebView.tsx:157]:
```suggestion
				onLoadEnd={() => setLoading(false)}
```

File: app/views/AuthenticationWebView.tsx [hunk 5]
```diff
@@ -173,39 +132,35 @@ class AuthenticationWebView extends React.PureComponent<IAuthenticationWebView,
...
+	const isIframe = authType === 'iframe';
 
-const mapStateToProps = (state: IApplicationState) => ({
-	server: state.server.server,
-	Accounts_Iframe_api_url: state.settings.Accounts_Iframe_api_url as string,
-	Accounts_Iframe_api_method: state.settings.Accounts_Iframe_api_method as string
-});
+	useLayoutEffect(() => {
+		navigation.setOptions({
+			headerLeft: () => <HeaderButton.CloseModal navigation={navigation} />,
```
Comment (GleidsonDaniel) [inline app/views/AuthenticationWebView.tsx:139]:
```suggestion
			headerLeft: () => <HeaderButton.CloseModal />,
```
This component does not need to pass navigation

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

**Review by GleidsonDaniel — APPROVED** [review]

**Pull request merged**